### PR TITLE
Add crud menu logic

### PR DESCRIPTION
**Pull request by snorrekr**
This PR adds a CRUD system for lyche's menu.
The new meny consists of two models, sulten_menu_category and sulten_menu_item.

A category is for thematic grouping such as *starters*, *main courses*, *dessert* etc.
An item is a dish and is related to a category. A *burger* is an item under *main courses* 

Both category and item has an order field which is a number representing where on the menu it appears. 
Category-order is first and then item-order.

example: 
- Main courses, 2
    - Burger, 2
    - Fish, 1
- Starters, 1
    - Bread 1
   
Actuall menu:
Starters:
    Bread
Main courses:
    Fish
    Burger

### CODE REVIEW

**Comment by snorrekr** [comment]
I've'nt implemented order for items, will do now quickly.

File: app/assets/stylesheets/modern/button.scss [hunk 1]
```diff
@@ -22,59 +22,75 @@
...
   background-color: #eee;
   filter: brightness(100%);
 }
+
 .samf-button.black {
   color: white;
   background-color: black;
   border: 1px solid #444;
 }
+
```
Comment (robines) [inline app/assets/stylesheets/modern/button.scss]: Mange unødvendige blanke linjeendringer og endringer som kun endrer formattering. Kunne du ha fjernet de så blir PRen en smule mer oversiktelig?

Comment (snorrekr) [reply app/assets/stylesheets/modern/button.scss]: will do.

Comment (snorrekr) [reply app/assets/stylesheets/modern/button.scss]: Done

**Review by robines — DISMISSED** [review]

File: db/migrate/20241010195439_add_sulten_menu.rb [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+      t.text :description_no
+      t.text :description_en
+      t.string :allergens_no
+      t.string :allergens_en
+      t.string :additional_info_no, null: true
+      t.string :additional_info_en, null: true
+      t.string :recommendation, null: true
+      t.integer :price
+      t.integer :price_member
+      t.references :category, foreign_key: { to_table: :sulten_menu_categories }
```
Comment (snorrekr) [inline db/migrate/20241010195439_add_sulten_menu.rb:21]: It previously did not work because it said `foreign_keys` and not `foreign_key`
bruh

**Review by robines — APPROVED** [review]

**Pull request merged**